core/arch/arm/tee/pta_socket.c

```diff
@@ -239,19 +242,21 @@ static TEE_Result socket_ioctl(uint32_t instance_id, uint32_t param_types,
 	msg_params[0].u.value.c = params[0].value.a; /* handle */
 
 	/* buffer */
-	msg_params[1].attr = OPTEE_MSG_ATTR_TYPE_TMEM_INOUT;
-	msg_params[1].u.tmem.buf_ptr = pa;
-	msg_params[1].u.tmem.size = params[1].memref.size;
-	msg_params[1].u.tmem.shm_ref = cookie;
+	if (!msg_param_init_memparam(msg_params + 1, mobj, 0,
+				     params[1].memref.size, cookie,
+				     MSG_PARAM_MEM_DIR_INOUT))
+		return TEE_ERROR_BAD_STATE;
+
 	memcpy(va, params[1].memref.buffer, params[1].memref.size);
 
 	msg_params[2].attr = OPTEE_MSG_ATTR_TYPE_VALUE_INPUT;
 	msg_params[2].u.value.a = params[0].value.b; /* ioctl command */
 
 	res = thread_rpc_cmd(OPTEE_MSG_RPC_CMD_SOCKET, 3, msg_params);
-	if (msg_params[1].u.tmem.size <= params[1].memref.size)
-		memcpy(params[1].memref.buffer, va, msg_params[1].u.tmem.size);
-	params[1].memref.size = msg_params[1].u.tmem.size;
+	if (msg_param_get_buf_size(msg_params + 1) <= params[1].memref.size)
+		memcpy(params[1].memref.buffer, va,
+		       msg_param_get_buf_size(msg_params + 1));
+	params[1].memref.size = msg_param_get_buf_size(msg_params + 1);
```

Comment:
Any particular reason why this function doesn't use the same pattern to update `params[1].memref` as `socket_recv()` above?